pkg/kubenest/common/resource.go

```diff
@@ -0,0 +1,14 @@
+package common
+
+import (
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kosmos.io/kosmos/pkg/apis/kosmos/v1alpha1"
+)
+
+type Resource struct {
+	Namespace     string
+	Name          string
+	Vc            *v1alpha1.VirtualCluster
+	RootClientSet kubernetes.Interface
```

Comment:
Since information from the Resource struct is needed, a struct is defined here and placed in the common package for use by multiple Go files. However, the struct name can be changed to better reflect its purpose.